01/table_making/1F.hector.table.R

```diff
@@ -0,0 +1,163 @@
+# Title: Remaking Manuscript Table 3
+# Author: Peter Scully
+# Date: 6/5/24
+
+# Import statements
+library(hector)
+library(dplyr)
+
+# Storing output file path
+OUTPUT <- file.path(here::here(), "01", "table_making", "table_replica.txt")
+
+#################
+### FUNCTIONS ###
+#################
+
+
+# run_hector - function to run Hector using a given ini file
+#
+# args:
+#   ini_file - path to the ini file to use to run Hector
+#   yrs      - year range to get Hector data from
+#   vars     - Hector variables to get data on
+#
+# returns: data frame containing the variables' values for the given date range
+run_hector <- function(ini_file, yrs, vars) {
+  core <- newcore(ini_file)
+  run(core)
+  data <- fetchvars(core, yrs, vars = vars)
+  shutdown(core)
+  return(data)
+}
+
+
+# get_interval_avg - function to find the average value of a Hector variable
+#                    over a given time interval
+#
+# args:
+#   data  - data frame outputted by run_hector (or fetchvars)
+#   var   - string containing the Hector variable to average
+#   start - start year for finding average
+#   end   - end year for finding average
+#
+# returns: average value of var across the provided interval
+get_interval_avg <- function(data, var, start, end) {
+  data %>%
+    subset(variable == var) %>%
+    subset(start <= year & year <= end) %>%
+    .$value -> interval_vals
+  return(mean(interval_vals))
+}
+
+
+# rel_to_val - function to adjust values of a Hector variable to be relative
+#              to a new value
+# args:
+#   data      - data frame outputted by fetchvars
+#   var       - string containing the Hector variable to adjust
+#   benchmark - value to make variable relative to
+#
+# returns: data frame containing the new values for var relative to benchmark
+rel_to_val <- function(data, var, benchmark) {
+  # Updating all of the values for var that are in data
+  data %>%
+    mutate(value = ifelse(variable == var, value - benchmark, value)) -> data
+
+  return(data)
+}
+
+# rel_to_interval - function to normalize values of a Hector variable to a
+#                   reference period
+# args:
+#   data  - data frame outputted by fetchvars
+#   var   - string containing the Hector variable to adjust
+#   start - start year for reference period
+#   end   - end year for reference period
+#
+# returns - data frame containing the normalized values for that variable
+rel_to_interval <- function(data, var, start, end) {
+  benchmark <- get_interval_avg(data, var, start, end)
+  return(rel_to_val(data, var, benchmark))
```

Comment:
Could you add in a `signif(value, 3)`call before  you return the output? This should take care of rounding differences.

Reply:
I just tried adding in signif, but it didn't fix the off-by-0.01 errors.